tests/spec/date_series_ops/test_date_comparisons.py

```diff
@@ -183,3 +185,149 @@ def test_is_not_in_strings(spec_test):
             4: None,
         },
     )
+
+
+table_data_for_between_tests = {
+    p: """
+          |     d1
+        --+------------
+        1 | 2010-01-01
+        2 | 2010-01-02
+        3 | 2010-01-03
+        4 | 2010-01-04
+        5 | 2010-01-05
+        6 |
+        """,
+}
+
+
+def test_is_between_dates(spec_test):
+    table_data = table_data_for_between_tests
+    spec_test(
+        table_data,
+        p.d1.is_between(date(2010, 1, 2), date(2010, 1, 4)),
+        {
+            1: False,
+            2: False,
+            3: True,
+            4: False,
+            5: False,
+            6: None,
+        },
+    )
+
+
+def test_is_on_or_between_dates(spec_test):
+    table_data = table_data_for_between_tests
+    spec_test(
+        table_data,
+        p.d1.is_on_or_between(date(2010, 1, 2), date(2010, 1, 4)),
+        {
+            1: False,
+            2: True,
+            3: True,
+            4: True,
+            5: False,
+            6: None,
+        },
+    )
+
+
+def test_is_between_strings(spec_test):
+    table_data = table_data_for_between_tests
+    spec_test(
+        table_data,
+        p.d1.is_between("2010-01-02", "2010-01-04"),
+        {
+            1: False,
+            2: False,
+            3: True,
+            4: False,
+            5: False,
+            6: None,
+        },
+    )
+
+
+def test_is_on_or_between_strings(spec_test):
+    table_data = table_data_for_between_tests
+    spec_test(
+        table_data,
+        p.d1.is_on_or_between("2010-01-02", "2010-01-04"),
+        {
+            1: False,
+            2: True,
+            3: True,
+            4: True,
+            5: False,
+            6: None,
+        },
+    )
+
+
+def test_is_between_dates_backwards(spec_test):
+    table_data = table_data_for_between_tests
+    with pytest.raises(AssertionError):
```

Comment:
The error you're getting from the docs generator is due to this `pytest.raises` construction, rather than anything to do with the `table_data_for_between_tests` trick.

The spec tests don't currently have a mechanism for giving examples of things you're _not_ allowed to do (though it has been [discussed](https://github.com/opensafely-core/databuilder/issues/521) before).

In this particular case I think the answer is to keep the examples, but change the expected results from `True` to `False`. That at least demonstrates that the arguments are sensitive to order.

Reply:
Sure, that seems good to me